### PR TITLE
[Memory64] preamble JS changes for 64-bit

### PR DESCRIPTION
**Pull request by aardappel**

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -749,6 +752,50 @@ function instrumentWasmTableWithAbort() {
...
+// BigInts. For now we keep JS as much the same as it always was, that is,
+// stackAlloc() receives and returns a Number from the JS point of view -
+// we translate BigInts automatically for that.
+function instrumentWasmExportsForMemory64(exports) {
+  var instExports = {};
+  for (var name in exports) {
+    (function(name) {
+      var original = exports[name];
+      var replacement = original;
+      if (name === 'stackAlloc' || name === 'malloc') {
```
Comment (kripken) [inline src/preamble.js:766]: This method won't work if we minify the names of exports, which we do in `-O3` etc. For now, please add a comment `// TODO: support minified export names`, and in a later PR we should disable that minification for `MEMORY64` for now (that would be in `minify_wasm_imports_and_exports` in `tools/building.py`)

**Review by kripken — APPROVED** [review]
lgtm % comment

File: src/preamble.js (hunk 1 above)
Comment (aardappel) [reply src/preamble.js:767]: Ok, will do.

Comment (aardappel) [reply src/preamble.js:767]: Addressed here: https://github.com/emscripten-core/emscripten/pull/15305

**Pull request merged**